### PR TITLE
[pigeon] added support for builtin java datatypes

### PR DESCRIPTION
**Pull request by gaaclarke**
depends on: https://github.com/flutter/packages/pull/96

Now things like ints and data buffers can be used in classes, not just Strings.

### CODE REVIEW

**Comment by xster** [comment]
This is the next one now?

File: packages/pigeon/lib/generator_tools.dart [hunk 1]
```diff
@@ -82,3 +82,37 @@ class Indent {
...
+
+/// Represents the mapping of a Dart datatype to a Host datatype.
+class HostDatatype {
+  /// Parametric constructor for HostDatatype.
+  HostDatatype({this.datatype, this.isBuiltin});
+
+  /// The [String] that can be printed into host code to represent the type.
+  final String datatype;
+
+  /// `true` if the host datatype is something builtin.
```
Comment (xster) [inline packages/pigeon/lib/generator_tools.dart]: Still not sure what this means. Does this mean primitive to Java? Or a Dart SDK class in Dart?

File: packages/pigeon/lib/generator_tools.dart [hunk 2]
```diff
@@ -82,3 +82,37 @@ class Indent {
...
+  final bool isBuiltin;
+}
+
+/// Calculates the [HostDatatype] for the provided [Field].  It will check the
+/// field against the [classes] to check if it is a builtin type.
```
Comment (xster) [inline packages/pigeon/lib/generator_tools.dart]: if you're referring to arguments, the dartdoc should be `classes`

File: packages/pigeon/lib/generator_tools.dart (hunk 1 above)
Comment (xster) [reply packages/pigeon/lib/generator_tools.dart]: ohhh wait, this is the builtin handled types for our codec?

Comment (gaaclarke) [reply packages/pigeon/lib/generator_tools.dart]: Right, as opposed to types defined as collections of supported types.  These are the atomic types.

File: packages/pigeon/lib/generator_tools.dart (hunk 2 above)
Comment (gaaclarke) [reply packages/pigeon/lib/generator_tools.dart]: done

**Pull request merged**